app/controllers/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!, only: %i[fight add_experience]
+
+  def fight
+    @user = current_user
+    @bot = @user.dup
+    @bot.hp = rand(@user.hp * 0.8..@user.hp + @user.hp * 0.2)
```

Comment:
what is `hp` ? Change the variable name to more understandable.

Reply:
If a new person starts working with your code, it will be hard to find out what you mean here

Reply:
ok